### PR TITLE
feat: Adding Docker setup for RL

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

Adding Docker recipe + associated scripts for running RL agents

### Testing

Ran locally, appears to be all working

### Breaking Changes

None

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,5 +34,18 @@ COPY ./examples ./examples
...
+RUN chmod 777 /vega_market_sim
 
-USER vega
\ No newline at end of file
+USER vega
+
+FROM vegasim_base AS vegasim_learning
+
+COPY ./requirements-learning.txt .
+RUN  pip install -r requirements-learning.txt
```
Comment (cdummett) [inline Dockerfile:44]: More my own interest than feedback - any reason for using pip over Poetry for the docker build?

**Review by cdummett — COMMENTED** [review]
Testing build and learning scripts now 👍

**Comment by cdummett** [comment]
@TomMcL Am now failing to build the image with this error message...

```
 => ERROR [vegasim_learning 4/5] RUN pip install -e .                                                                                            92.3s
------
 > [vegasim_learning 4/5] RUN pip install -e .:
#0 2.240 Obtaining file:///vega_market_sim
#0 2.259   Installing build dependencies: started
#0 71.67   Installing build dependencies: still running...
#0 81.99   Installing build dependencies: finished with status 'error'
#0 82.03   error: subprocess-exited-with-error
#0 82.03   
#0 82.03   × pip subprocess to install build dependencies did not run successfully.
#0 82.03   │ exit code: 1
#0 82.03   ╰─> [8 lines of output]
#0 82.03       WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x40040827d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/poetry-core/
#0 82.03       WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4004082b00>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/poetry-core/
#0 82.03       WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x40040838e0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/poetry-core/
#0 82.03       WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4004083a90>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/poetry-core/
#0 82.03       WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4004083c40>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/poetry-core/
#0 82.03       ERROR: Could not find a version that satisfies the requirement poetry-core>=1.0.0 (from versions: none)
#0 82.03       ERROR: No matching distribution found for poetry-core>=1.0.0
#0 82.03       WARNING: There was an error checking the latest version of pip.
#0 82.03       [end of output]
#0 82.03   
#0 82.03   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 82.04 error: subprocess-exited-with-error
#0 82.04 
#0 82.04 × pip subprocess to install build dependencies did not run successfully.
#0 82.04 │ exit code: 1
#0 82.04 ╰─> See above for output.
#0 82.04 
#0 82.04 note: This error originates from a subprocess, and is likely not a problem with pip.
#0 92.17 WARNING: There was an error checking the latest version of pip.
------
ERROR: failed to solve: executor failed running [/bin/sh -c pip install -e .]: exit code: 1
Charlies-MacBook-Pro:vega-market-sim caed$ 
```

File: Dockerfile (hunk 1 above)
Comment (TomMcL) [reply Dockerfile:44]: Ultimately Poetry is just overkill in a Docker environment. You're only building the once then throwing the thing away so you can just install your python packages to system python and be done with it without the weight of venvs etc

**Review by cdummett — APPROVED** [review]
LGTM

**Pull request merged**